app/code/community/Bolt/Boltpay/controllers/ShippingController.php

```diff
@@ -103,6 +97,12 @@ public function indexAction()
             } else {
                 Mage::helper('boltpay')->setCustomerSessionById($quote->getCustomerId());
```

Comment:
Good insight.  The fix here is setting the customer id to the session prior to setting the quote id to the session.

Instead of moving the block of code for setting the quote id to the session, please move this line of code with regards to setting the customer id to the session to above the original position of session quote id being set to the session (i.e. before line 71).

Both the customer id and the quote id need to be set here as they are used in both portions of the succeeding `if\else` block.  Otherwise, you introduce the possibility of errors when this code is reached:

```
                     $response = $shippingAndTaxModel->getShippingAndTaxEstimate($quote);
                     $cacheBoltHeader = 'MISS';	                     $cacheBoltHeader = 'MISS';
```